projects/igniteui-angular/src/lib/grids/common/enums.ts

```diff
@@ -34,6 +34,12 @@ export const GridSelectionMode = mkenum({
 });
 export type GridSelectionMode = (typeof GridSelectionMode)[keyof typeof GridSelectionMode];
 
+export const HierarchicalGridSelectionMode = mkenum({
```

Comment:
Don't introduce a new enum, just add the new selection mode to the `GridSelectionMode`. If anyone sets it on a flat or hierarchical grid, then selection would still work as multiple, because the `isMultiRowSelectionEnabled` would still return true, even though there's no cascading in flat and hiearchical grids.

Reply:
Technically, the 'enum' is only there for compatibility with older code when those options were actual enums and for this specifically, it's not even needed but likely better to have for consistency. The selection property (and others we've converted) are simple string union types now - their usage in no way requires the exported constants like this one as they are not even necessary. Just like in markup, this is perfectly valid:
```ts
this.grid1.rowSelection = 'multiple';
```
There's intellisense autocomplete, it's type checked (i.e. you can't just assign a random string) and will even validate conditions like this:
![image](https://user-images.githubusercontent.com/3198469/108046768-acdeb000-704d-11eb-92c7-3880e3274ce6.png)

It costs us very little override the `'single' | 'multiple' | 'multipleCascade'` with `'single' | 'multiple'` for the flat grid and I'd rather do that to avoid the API exposing something it doesn't support even if it would technically still work. There's no impact on existing code, even for the tree grid you can still assign values from the old constant too.
Again, the exported constants are only there for backwards compatibility and/or [questionable] convenience and the selection pre-dated the migration, so the new one is there to be consistent, though for new 'enums' those constants are not even needed to be publicly exported and can be kept just for internal references.

Reply:
@damyanpetev This is unnecessary code pollution, which adds no value to the actual features. All it does for us is increase the side of the bundle.

Reply:
Yes, no value added for the feature itself, the only benefit was keeping the public API of the unaffected grids clean. We were more concerned with that, rather than the duplicating one get/set (the H Grid override is supposed to be temporary). However, I see your point about pollution and duplication and I agree it's unnecessary debt and a small bump to the packed size.

Not sure if unreasonable, but the duplication can be avoided if the base is generic on the selection mode, and this I think can be implemented fairly gracefully with very little changes overall by making it optional as well. Basically, just convincing TypeScript to show the correct type, while keeping all the code the same.

If you think that's overkill, then a comment on the property the `multipleCascade` only applies to the Tree Grid (or doesn't apply to flat and H grids) will have to do.